### PR TITLE
docs(zh-CN): add References -> Users -> Profile -> Basic data

### PR DESCRIPTION
**Pull request by IceHe**
Add the zh-CN version of References -> Users -> Profile -> Basic data

### CODE REVIEW

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 1]
```diff
@@ -79,3 +79,91 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+
+#### application id
+
+_application id_ 即用户首次注册时对应的 [应用 ID](../applications/#应用-id)（_client id_）。
+
+#### last signed in at
+
+_last signed in at_ 即用户上次登录时的带时区的时间戳。
+
+#### password encrypted
```
Comment (gao-sun) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:144]: this enlightens me to add a section "Security" in docs. would you like to add an issue in the Docs v1.1 linear project? thanks

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 2]
```diff
@@ -79,3 +79,91 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+如果 _password encrypted_ 的值非空，则对应加密前的明文密码包含至少 6 个字符。
+
+#### password encryption method
+
+_password encryption method_ 即用户密码的加密方式。
+当用户没用过用户名加密码的方式注册时，其 _password encryption method_ 可能是空的。
+
+Logto 默认使用 [Argon2](https://en.wikipedia.org/wiki/Argon2) 的实现 [node-argon2](https://github.com/ranisalt/node-argon2) 作为加密方式；如果你有兴趣，可以参阅链接的资料以了解细节。
+
+`password_encrypted` 和 `password_encryption_method` 的示例，明文密码为“123456”：
```
Comment (gao-sun) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:158]:
```suggestion
`password_encrypted` 和 `password_encryption_method` 的示例，明文密码为 `123456`：
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 3]
```diff
@@ -79,3 +79,91 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+
+:::info
+
+例如，只有 _roles names_ 中包含 _admin_ 的用户，即管理员用户，才有权限使用 Logto 中的「管理控制台」和「管理 API」。
+
+:::
+
+#### application id
+
+_application id_ 即用户首次注册时对应的 [应用 ID](../applications/#应用-id)（_client id_）。
```
Comment (gao-sun) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:138]: should mention client id is an OIDC concept

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 4]
```diff
@@ -79,3 +79,91 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+
+如果用户使用 Facebook 和微信等社交连接器注册，则可能会直接采用社交用户信息中获取到的头像。
+允许的图片 URL 最多 2048 个字符。
+
+#### role names
+
+_roles names_ 即用户的角色名称集合，表示在 Logto 中授予给该用户的角色集合。
+
+_role_ 即角色，代表你授予给用户的一组权限。
+与单独为用户分配一个个权限相比，使用角色更便于授予、撤销和调整用户的权限。
```
Comment (gao-sun) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:128]: should we mention currently there's no way to update this in admin console?

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md (hunk 1 above)
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:144]: ok. LOG-3386

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md (hunk 3 above)
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:138]: updated. PTAL

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md (hunk 4 above)
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:128]: updated. PTAL

```markdown
You cannot update the user's _role names_ using Admin Console for now. Stay tuned.

你目前还不能通过「管理控制台」更新用户的 _role names_，敬请关注。
```

File: docs/docs/references/users/README.md [hunk 5]
```diff
@@ -9,8 +9,8 @@ Each user has a profile containing the user information.
 
 A user profile consists of the following types of data:
 
-- [**Basic data**](#basic-data): contains basic info in pre-defined attributes, such as user id, username, email, phone number, and when the user last signed in.
-- [**Custom data**](#custom-data): contains additional info in customizable attributes, such as user-preferred color and language.
+- [**Basic Data**](#basic_data): contains basic info in pre-defined attributes, such as user id, username, email, phone number, and when the user last signed in.
+- [**Custom Data**](#custom_data): contains additional info in customizable attributes, such as user-preferred color and language.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:13]:
```suggestion
- [_basic_data_](#basic_data): contains basic info in pre-defined attributes, such as user id, username, email, phone number, and when the user last signed in.
- [_custom_data_](#custom_data): contains additional info in customizable attributes, such as user-preferred color and language.
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 6]
```diff
@@ -79,3 +79,93 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+允许的图片 URL 最多 2048 个字符。
+
+#### role_names
+
+roles_names 即用户的角色名称集合，表示在 Logto 中授予给该用户的角色集合。
+
+role 即角色，代表你授予给用户的一组权限。
+与单独为用户分配一个个权限相比，使用角色更便于授予、撤销和调整用户的权限。
+
+你目前还不能通过「管理控制台」更新用户的 role_names，敬请关注。
```
Comment (gao-sun) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:130]:
```suggestion
你目前还不能通过「管理控制台」更新用户的 role_names。
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 7]
```diff
@@ -79,3 +79,93 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+avatar 即用户的头像，是指向用户头像图片的 URL。
+
+如果用户使用 Facebook 和微信等社交连接器注册，则可能会直接采用社交用户信息中获取到的头像。
+允许的图片 URL 最多 2048 个字符。
+
+#### role_names
+
+roles_names 即用户的角色名称集合，表示在 Logto 中授予给该用户的角色集合。
+
+role 即角色，代表你授予给用户的一组权限。
```
Comment (gao-sun) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:127]: "role" should not appear in this section.

File: docs/docs/references/users/README.md [hunk 8]
```diff
@@ -9,8 +9,8 @@ Each user has a profile containing the user information.
...
 - [**Identities**](#identities): contains the user info retrieved from social sign-in (i.e., sign-in with a social connector), such as Facebook, GitHub, and WeChat.
```
Comment (IceHe) [inline docs/docs/references/users/README.md:14]:
```suggestion
- [_identities_](#identities): contains the user info retrieved from social sign-in (i.e., sign-in with a social connector), such as Facebook, GitHub, and WeChat.
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 9]
```diff
@@ -79,3 +79,93 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+与单独为用户分配一个个权限相比，使用角色更便于授予、撤销和调整用户的权限。
+
```
Comment (IceHe) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:129]:
```suggestion
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 10]
```diff
@@ -79,3 +79,93 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+#### avatar
+
+avatar 即用户的头像，是指向用户头像图片的 URL。
+
+如果用户使用 Facebook 和微信等社交连接器注册，则可能会直接采用社交用户信息中获取到的头像。
+允许的图片 URL 最多 2048 个字符。
+
+#### role_names
+
+roles_names 即用户的角色名称集合，表示在 Logto 中授予给该用户的角色集合。
```
Comment (IceHe) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:125]:
```suggestion
roles_names 即用户的角色集合。
```

File: docs/docs/references/users/README.md [hunk 11]
```diff
@@ -125,38 +125,40 @@ _avatar_ is the URL pointing to the user's avatar image.
...
 Its max length is 2048.
 
-#### role names
+#### role_names
 
-_roles names_ represent the roles granted to the user in Logto.
+roles_names represent the roles granted to the user in Logto.
 
 A _role_ is a group of permissions that you can grant to users.
 Using roles makes it easier to grant, revoke, and adjust permissions than assigning permissions to users individually.
```
Comment (IceHe) [inline docs/docs/references/users/README.md:134]:
```suggestion
A _role_ is a group of permissions that you can grant to users.
Using roles makes it easier to grant, revoke, and adjust permissions than assigning permissions to users individually.

```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md (hunk 7 above)
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:127]: removed this paragraph.

File: docs/docs/references/users/README.md [hunk 12]
```diff
@@ -9,9 +9,9 @@ Each user has a profile containing the user information.
 
 A user profile consists of the following types of data:
 
-- [**Basic data**](#basic-data): contains basic info in pre-defined attributes, such as user id, username, email, phone number, and when the user last signed in.
-- [**Custom data**](#custom-data): contains additional info in customizable attributes, such as user-preferred color and language.
-- [**Identities**](#identities): contains the user info retrieved from social sign-in (i.e., sign-in with a social connector), such as Facebook, GitHub, and WeChat.
+- [_basic_data_](#basic_data): contains basic info in pre-defined attributes, such as user id, username, email, phone number, and when the user last signed in.
+- [_custom_data_](#custom_data): contains additional info in customizable attributes, such as user-preferred color and language.
+- [_identities_](#identities): contains the user info retrieved from social sign-in (i.e., sign-in with a social connector), such as Facebook, GitHub, and WeChat.
```
Comment (IceHe) [inline docs/docs/references/users/README.md:14]:
```suggestion
- [basic_data](#basic_data) contains basic info in pre-defined attributes, such as user id, username, email, phone number, and when the user last signed in.
- [custom_data](#custom_data) contains additional info in customizable attributes, such as user-preferred color and language.
- [identities](#identities) contains the user info retrieved from social sign-in (i.e., sign-in with a social connector), such as Facebook, GitHub, and WeChat.
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 13]
```diff
@@ -79,3 +79,90 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+roles_names 即用户的角色集合。
+
+你目前还不能通过「管理控制台」更新用户的 role_names。
+
+:::info
+
+例如，只有 roles_names 中包含 admin 的用户，即管理员用户，才有权限使用 Logto 中的「管理控制台」和「管理 API」。
+
+:::
+
```
Comment (IceHe) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:134]:
```suggestion
:::info

例如，只有 roles_names 中包含 admin 的用户，即管理员用户，才有权限使用 Logto 中的「管理控制台」和「管理 API」。

:::

你目前还不能通过「管理控制台」更新用户的 role_names。

```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 14]
```diff
@@ -79,3 +79,93 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+如果用户没用过用户名加密码的方式注册，他们的 _username_ 可能是空的。
+_username_ 的值非空时，最多 128 个字符，只能包含字母、数字和下划线（`_`），不能以数字开头。
+
+#### primary email
+
+_primary email_ 即主要邮箱，是用户的电子邮件地址，用于邮箱加验证码的登录方式。
+
+如果用户没用过邮箱注册，他们的 _primary email_ 的值可能是空的。
+_primary email_ 的值非空时，最多 128 个字符。
```
[the remark this answers is not in the document]
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md]: varchar(128) in DB attribute

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 15]
```diff
@@ -79,3 +79,93 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+_role_ 即角色，代表你授予给用户的一组权限。
+与单独为用户分配一个个权限相比，使用角色更便于授予、撤销和调整用户的权限。
+
+你目前还不能通过「管理控制台」更新用户的 _role names_，敬请关注。
+
+:::info
+
+例如，只有 _roles names_ 中包含 _admin_ 的用户，即管理员用户，才有权限使用 Logto 中的「管理控制台」和「管理 API」。
+
+:::
```
[the remark this answers is not in the document]
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md]: updated

File: docs/docs/references/users/README.md [hunk 16]
```diff
@@ -98,19 +98,19 @@ _username_ is used for sign-in with _username_ and password.
...
-#### primary email
+#### primary_email
 
-_primary email_ is the user's email address, used for sign-in with the email and passcode.
+primary_email is the user's email address, used for sign-in with the email and passcode.
 
-When the user has not registered with their email, their _primary email_ may be empty.
-When the _primary email_ is non-empty, it should be at most 128 characters.
+When the user has not registered with their email, their primary_email may be empty.
+When the primary_email is non-empty, it should be at most 128 characters.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:106]:
```suggestion
It should be no longer than 128 characters.
```

File: docs/docs/references/users/README.md [hunk 17]
```diff
@@ -98,19 +98,19 @@ _username_ is used for sign-in with _username_ and password.
...
-#### primary phone
+#### primary_phone
 
-_primary phone_ is the user's phone number, used for sign-in with the phone number and passcode from SMS.
+primary_phone is the user's phone number, used for sign-in with the phone number and passcode from SMS.
 
-When the user has not registered with their phone number, their _primary phone_ may be empty.
-When the _primary phone_ is non-empty, it should contain numbers prefixed with the
+When the user has not registered with their phone number, their primary_phone may be empty.
+When the primary_phone is non-empty, it should contain numbers prefixed with the
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:113]:
```suggestion
When it's non-empty, it should contain numbers prefixed with the
```

File: docs/docs/references/users/README.md [hunk 18]
```diff
@@ -125,38 +125,37 @@ _avatar_ is the URL pointing to the user's avatar image.
...
 
-#### role names
+#### role_names
 
-_roles names_ represent the roles granted to the user in Logto.
+roles_names represent the roles granted to the user in Logto.
 
-A _role_ is a group of permissions that you can grant to users.
-Using roles makes it easier to grant, revoke, and adjust permissions than assigning permissions to users individually.
+You cannot update the user's role_names using Admin Console for now. Stay tuned.
```
Comment (gao-sun) [inline docs/docs/references/users/README.md:132]:
```suggestion
You cannot update the user's role_names using Admin Console for now.
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 19]
```diff
@@ -79,3 +79,93 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
 你可以使用「[管理控制台](../../../docs/recipes/manage-users/using-admin-console#查看和更新用户资料)」或
 「[管理 API](../../../docs/recipes/manage-users/using-management-api)」来查询用户资料，
 例如 <a href="/api/#tag/Users/paths/~1api~1users~1:userId/get" target="_blank">`GET /api/users/:userId`</a>。
+
+### 基础数据
+
+用户的基础数据，即在预定义的属性中保存的用户基础信息。
```
[the remark this answers is not in the document]
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:85]: will remove this sentence, and keep their description in the Profile above

File: docs/docs/references/users/README.md [hunk 20]
```diff
@@ -98,19 +98,19 @@ _username_ is used for sign-in with _username_ and password.
 When the user has not registered with their username, their _username_ may be empty.
 When the _username_ is non-empty, it should be at most 128 characters, only contain letters, numbers, and underscores (`_`), and NOT start with a number.
 
-#### primary email
+#### primary_email
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:101]: should we mention that primary email/phone is verified?
it should not appear in these two fields if a phone number or email is not verified.

File: docs/docs/references/users/README.md [hunk 21]
```diff
@@ -189,27 +188,27 @@ Sample _custom data_ from an admin user in Logto:
 }
 ```
 
-Each user's _custom data_ is stored in an individual JSON object.
+Each user's custom_data is stored in an individual JSON object.
 
-:::caution Put sensitive data in _custom data_
+:::caution Put sensitive data in custom_data
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:193]:
```suggestion
:::caution DO NOT put sensitive data in custom_data
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 22]
```diff
@@ -8,29 +8,29 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
 
 一份 _用户资料_ 由以下几个类型的数据组成：
 
-- [**基础数据**](#基础数据basic-data)：在预定义的属性中保存基础信息，例如用户 ID、用户名、邮箱、手机号以及用户上次登录的时间。
-- [**自定义数据**](#自定义数据custom-data)：在可自定义的属性中保存附加信息，例如用户偏好的颜色和语言。
-- [**身份**](#身份identities)：包括从社交登录（即使用社交连接器登录）获取的用户信息，例如 Facebook、GitHub 和微信。
+- [**基础数据**](#basic_data)：在预定义的属性中保存基础信息，例如用户 ID、用户名、邮箱、手机号以及用户上次登录的时间。
+- [**自定义数据**](#custom_data)：在可自定义的属性中保存附加信息，例如用户偏好的颜色和语言。
+- [**身份**](#identities)：包括从社交登录（即使用社交连接器登录）获取的用户信息，例如 Facebook、GitHub 和微信。
```
Comment (darcyYe) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:13]: should we add in-page link here?

Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:13]: i think it's better

File: docs/docs/references/users/README.md (hunk 21 above)
Comment (IceHe) [reply docs/docs/references/users/README.md:190]: this fix is put in next PR. please ignore

File: docs/docs/references/users/README.md (hunk 20 above)
Comment (IceHe) [reply docs/docs/references/users/README.md:99]: Per Slack thread [discussion](https://silverhand-io.slack.com/archives/C02CMB8JVJB/p1656919713519819?thread_ts=1656835704.413199&cid=C02CMB8JVJB).

File: docs/docs/references/users/README.md [hunk 23]
```diff
@@ -83,13 +83,13 @@ You can query the user profile using
...
-### basic data
+### basic_data
 
-_basic data_ contains the basic user info in pre-defined attributes.
+basic_data contains the basic user info in pre-defined attributes.
 
 #### user id
 
-_user id_ is a unique auto-generated key to identify the user in Logto.
+user_id is a unique auto-generated key to identify the user in Logto.
```
[the remark this answers is not in the document]
Comment (IceHe) [reply docs/docs/references/users/README.md:90]: Updated per Slack thread [discussion](https://silverhand-io.slack.com/archives/C02CMB8JVJB/p1656835704413199).

File: docs/docs/references/users/README.md [hunk 24]
```diff
@@ -83,13 +83,13 @@ You can query the user profile using
 [Management API](../../../docs/recipes/manage-users/using-management-api),
 such as <a href="/api/#tag/Users/paths/~1api~1users~1:userId/get" target="_blank">`GET /api/users/:userId`</a>.
 
-### basic data
+### basic_data
 
-_basic data_ contains the basic user info in pre-defined attributes.
+basic_data contains the basic user info in pre-defined attributes.
```
[the remark this answers is not in the document]
Comment (IceHe) [reply docs/docs/references/users/README.md]: updated.

File: docs/docs/references/users/README.md [hunk 25]
```diff
@@ -189,27 +185,27 @@ Sample _custom data_ from an admin user in Logto:
 }
 ```
 
-Each user's _custom data_ is stored in an individual JSON object.
+Each user's _custom_data_ is stored in an individual JSON object.
 
-:::caution Put sensitive data in _custom data_
+:::caution DO NOT Put sensitive data in _custom_data_
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:190]:
```suggestion
:::caution DO NOT put sensitive data in _custom_data_
```

File: docs/docs/references/users/README.md [hunk 26]
```diff
@@ -83,34 +83,32 @@ You can query the user profile using
...
 
-#### primary email
+#### primary_email
 
-_primary email_ is the user's email address, used for sign-in with the email and passcode.
+_primary_email_ is the user's email address, used for sign-in with the email and passcode.
 
-When the user has not registered with their email, their _primary email_ may be empty.
-When the _primary email_ is non-empty, it should be at most 128 characters.
+When the user has not registered with their email, its value may be empty.
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:103]:
```suggestion
Its value should be `null` if the user did not register by email.
```

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md [hunk 27]
```diff
@@ -79,3 +79,89 @@ _用户_ 是身份服务的主要实体。我们将会在下文描述用户相
...
+_primary_email_ 即主要邮箱，是用户的电子邮件地址，用于邮箱加验证码的登录方式。
+
+如果用户没用过邮箱注册，该值可能是空的。最多 128 个字符。
+
+#### primary_phone
+
+_primary_phone_ 即主要手机号，用于手机号加短信验证码的登录方式。
+
+如果用户没用过手机号注册时，该值可能是空的。
+该值非空时，只能包含数字，且以 [国家呼叫代码](https://en.wikipedia.org/wiki/List_of_country_calling_codes) 为前缀（加号 `+` 除外）。
```
Comment (darcyYe) [inline i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:109]: nit: 国家或地区代码
image

**Review by darcyYe — APPROVED** [review]
overall lgtm

File: i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md (hunk 27 above)
Comment (IceHe) [reply i18n/zh-cn/docusaurus-plugin-content-docs/current/docs/references/users/README.md:109]: According to ZH Wikipedia, should use [国家地区编码](https://zh.wikipedia.org/zh-sg/%E5%9C%8B%E5%AE%B6%E5%9C%B0%E5%8D%80%E4%BB%A3%E7%A2%BC)

File: docs/docs/references/users/README.md [hunk 28]
```diff
@@ -83,34 +83,35 @@ You can query the user profile using
...
 
 #### username
 
 _username_ is used for sign-in with _username_ and password.
 
-When the user has not registered with their username, their _username_ may be empty.
-When the _username_ is non-empty, it should be at most 128 characters, only contain letters, numbers, and underscores (`_`), and NOT start with a number.
+Its value is from the username that the user first registered with.
+Its value may be `null`.
+Its non-null value should be no longer than 128 characters, only contain letters, numbers, and underscores (`_`), and NOT start with a number.
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:98]:
```suggestion
Its non-null value should not exceed 128 characters, only contain letters, numbers, and underscores (`_`), and NOT start with a number.
```

File: docs/docs/references/users/README.md [hunk 29]
```diff
@@ -121,46 +122,46 @@ Its max length is 128.
 #### avatar
 
 _avatar_ is the URL pointing to the user's avatar image.
-
-If the user registers with a social connector like Facebook and WeChat, their _avatar_ may be retrieved from the social user info.
 Its max length is 2048.
 
-#### role names
+If the user registers with a social connector like Facebook and WeChat, its value may be retrieved from the social user info.
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:127]:
```suggestion
If the user registers with a social media account like Facebook and WeChat, the `avatar` will be retrieved from the social user info.
```

File: docs/docs/references/users/README.md [hunk 30]
```diff
@@ -121,46 +122,46 @@ Its max length is 128.
...
+#### role_names
 
-A _role_ is a group of permissions that you can grant to users.
-Using roles makes it easier to grant, revoke, and adjust permissions than assigning permissions to users individually.
+_roles_names_ represent the roles granted to the user in Logto.
 
 :::info
 
-For example, only the users whose roles names contain `admin`, i.e., the admin users, have permission to use Admin Console and Management API in Logto.
+For example, only the users whose _roles_names_ contain `admin`, i.e., the admin users, have permission to use Admin Console and Management API in Logto.
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:135]:
```suggestion
For example, only the users whose _roles_names_ contains `admin`, i.e., the admin users, have permission to use "Admin Console" and Management API in Logto.
```

File: docs/docs/references/users/README.md [hunk 31]
```diff
@@ -189,27 +189,27 @@ Sample _custom data_ from an admin user in Logto:
...
 
-Each user's _custom data_ is stored in an individual JSON object.
+Each user's _custom_data_ is stored in an individual JSON object.
 
-:::caution Put sensitive data in _custom data_
+:::caution DO NOT put sensitive data in _custom_data_
 
-You may fetch a user profile as a whole containing _custom data_ using Management API and send it to the frontend apps or external backend services.
-Therefore, putting the sensitive information in _custom data_ may cause data leaks.
+You may fetch a user profile as a whole containing _custom_data_ using Management API and send it to the frontend apps or external backend services.
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:196]:
```suggestion
You may fetch a user profile containing _custom_data_ using Management API and send it to the frontend apps or external backend services.
```

File: docs/docs/references/users/README.md [hunk 32]
```diff
@@ -189,27 +189,27 @@ Sample _custom data_ from an admin user in Logto:
...
+Therefore, putting the sensitive information in _custom_data_ may cause data leaks.
 
-If you still want to put the sensitive information in _custom data_, we recommend encrypting it first.
+If you still want to put the sensitive information in _custom_data_, we recommend encrypting it first.
 Only encrypt/decrypt it in a trusted party like your backend services, and avoid doing it in the frontend apps.
-These will minimize the damage if your users' _custom data_ is leaked by mistake.
+These will minimize the damage if your users' _custom_data_ is leaked by mistake.
```
Comment (darcyYe) [inline docs/docs/references/users/README.md:201]: damage is more like a `physical` situation.

```suggestion
These will minimize the loss if your users' _custom_data_ is leaked by mistake.
```

**Review by darcyYe — APPROVED** [review]
overall LGTM, you may also fix the Chinese part accordingly.

File: docs/docs/references/users/README.md (hunk 28 above)
Comment (IceHe) [reply docs/docs/references/users/README.md:98]: just keep it.

File: docs/docs/references/users/README.md (hunk 29 above)
Comment (IceHe) [reply docs/docs/references/users/README.md:127]: just keep it

File: docs/docs/references/users/README.md (hunk 30 above)
Comment (IceHe) [reply docs/docs/references/users/README.md:135]: Updated.

**Pull request merged**